src/wallet/wallet.cpp

```diff
@@ -2918,6 +2918,23 @@ bool CWallet::AttachChain(const std::shared_ptr<CWallet>& walletInstance, interf
     assert(!walletInstance->m_chain || walletInstance->m_chain == &chain);
     walletInstance->m_chain = &chain;
 
+    // Unless allowed, ensure wallet files are not reused across chains:
+    if (!gArgs.GetBoolArg("-walletcrosschain", DEFAULT_WALLETCROSSCHAIN))
+    {
+        WalletBatch batch(walletInstance->GetDatabase());
+        CBlockLocator locator;
+        if (batch.ReadBestBlock(locator) && locator.vHave.size() > 0 && chain.getHeight())
+        {
+            // Wallet is assumed to be from another chain, if genesis block in the active
+            // chain is differs from the genesis block know to the wallet.
+            if (chain.getBlockHash(0) != locator.vHave.back())
+            {
+                error = _("Wallet files should not be reused across chains.");
```

Comment:
should this mention the setting to disable the check? Also, could go with `Untranslated()`? It is not like this is something that happens in everyday operation?

Reply:
Added a sentence mentioning `-walletcrosschain` and wrapped with `Untranslated(...)`.